pipelines/common/src/main/java/com/google/fhir/analytics/view/ViewDefinition.java

```diff
@@ -15,44 +15,169 @@
  */
 package com.google.fhir.analytics.view;
 
+import com.google.common.base.Strings;
+import com.google.common.collect.Sets;
 import com.google.gson.Gson;
 import java.io.IOException;
 import java.io.Reader;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import lombok.Getter;
 
 // TODO: Generate this class from StructureDefinition using tools like:
 //  https://github.com/hapifhir/org.hl7.fhir.core/tree/master/org.hl7.fhir.core.generator
-@Getter
 public class ViewDefinition {
 
-  private String name;
-  private String resource;
-  private String resourceVersion;
-  private List<Select> select;
-  private List<Where> where;
+  private static Pattern CONSTANT_PATTERN = Pattern.compile("%[A-Za-z][A-Za-z0-9_]*");
+
+  @Getter private String name;
+  @Getter private String resource;
+  @Getter private String resourceVersion;
+  @Getter private List<Select> select;
+  @Getter private List<Where> where;
+  // We don't need to expose constants because we do the replacement as part of the setup.
   private List<Constant> constant;
+  // This is also used internally for processing constants and should not be exposed.
+  Map<String, String> constMap = new HashMap<>();
 
-  public static ViewDefinition createFromFile(String jsonFile) throws IOException {
+  public static ViewDefinition createFromFile(String jsonFile)
+      throws IOException, ViewDefinitionException {
     Gson gson = new Gson();
     Path pathToFile = Paths.get(jsonFile);
     try (Reader reader = Files.newBufferedReader(pathToFile, StandardCharsets.UTF_8)) {
-      return gson.fromJson(reader, ViewDefinition.class);
+      ViewDefinition view = gson.fromJson(reader, ViewDefinition.class);
+      view.validateAndSetUp();
```

Comment:
During validation of the view, instead of breaking even when one validation fails, Can we collect all the failures and respond with all the errors. Probably the user experience might be good.

Reply:
Yes, I think this make sense for authoring tools.